awx/api/serializers.py

```diff
@@ -2201,6 +2199,64 @@ def create(self, validated_data):
         return return_data
 
 
+class BulkHostDeleteSerializer(serializers.Serializer):
+    def validate(self, attrs):
+        request = self.context.get('request', None)
+        # Extract the hosts ids from the querystring
+        d = request.query_params.get('pk_ids', '')
+        if d == '':
+            raise serializers.ValidationError({"Error": 'No hosts ids to delete'})
+        ids = [int(pk) for pk in d.split(',')]
+        if len(ids) > settings.BULK_HOST_MAX_DELETE:
+            raise serializers.ValidationError({"Error": 'Number of hosts exceeds system setting BULK_HOST_MAX_DELETE'})
+
+        # Getting list of all host objects, filtered by the list of the hosts to delete
+        attrs['in_inv'] = Host.objects.get_queryset().filter(pk__in=ids)
```

Comment:
might use `prefetch_related('inventory')` this may save some queries
example: https://github.com/ansible/awx/blob/8feeb5f1fae951332ea4b520e5618982109da570/awx/main/scheduler/task_manager.py#L98